src/pages/subscriptions-redirect.js

```diff
@@ -24,9 +23,7 @@ class SubscriptionsRedirectPage extends React.Component {
   }
 
   async componentDidMount() {
-    const { getLoggedInUser } = this.props;
-    const LoggedInUser = await getLoggedInUser();
-    this.setState({ LoggedInUser });
+    const { LoggedInUser } = this.props;
 
     if (!LoggedInUser) {
```

Comment:
When the page is loaded directly `LoggedInUser` is always going to be null in `componentDidMount` (the mount trigger the fetch), so this check is always going to fail and will redirect user to the signin page.

You can rely on `loadingLoggedInUser` to know if the current user in loading. That and a clever user of `componentDidMount` + `componendDidUpdate` should do the job :)

Reply:
Look again, please. I did a separated method that will be called by **componentDidMount** + **componentDidUpdate**. Is it right?
Another point: page events.js has similar behavior. I think that we also need to change there. What do you think?

Reply:
@Betree I also made a change for event.js, take a look, please.